workloopR tech note (#551)
ropensci/roweb2: content/technotes/2019-11-14-workloopR-release.Rmd

  ggtitle("Work loop plot via ggplot2") +
  theme_minimal()
```

See our [Plotting vignette](https://docs.ropensci.org/workloopR/articles/Plotting-workloopR.html) for more plotting ideas.

#### Summarizing a set of trials

The batch processing capabilities of workloopR should also be handy for efficiently analyzing all files within a specific folder, e.g., a set of trials from a single experiment. Here's an example:

```{r batch_processing}
## batch read and analyze a set of work loop trials stored within one directory
wl_batch_analyzed <- read_analyze_wl_dir(
                        system.file("extdata/wl_duration_trials",
                                    package = 'workloopR'),
                        sort_by = 'file_id',
                        phase_from_peak = TRUE,
                        cycle_def = 'lo',
                        GR = 2,
                        keep_cycles = 3
                     )

## now create a summary of the trials
wl_batch_summarized <- summarize_wl_trials(wl_batch_analyzed)
wl_batch_summarized

```

For more, see our [Batch processing vignette](https://docs.ropensci.org/workloopR/articles/batch-processing.html).

### How to get workloopR

> We are not (yet) on CRAN but the package is available through rOpenSci's github:

Reviewer: ```suggestion
We are not (yet) on CRAN but the package is available through rOpenSci's GitHub:
```
GitHub capitalization

workloopR, as the name implies, was originally designed to handle data from work loop experiments as well as experiments that are complementary to work loops like twitch and tetanic trials. 

We are happy to expand the scope of the package to incorporate even more types of analyses of muscle physiology or biomechanics. We're presently eager to add data import functions for non-`ddf` file types. We're also interested in integrating support for electromyographic ([EMG](https://en.wikipedia.org/wiki/Electromyography)) data recorded directly from muscles, but that may take a little longer to develop.

Should you like to suggest a specific feature, please use the [Issues page](https://github.com/ropensci/workloopR/issues/new/choose) of our repository. 

### Package review by rOpenSci and JOSS

We are thankful for the suggestions we've already received. workloopR benefited a lot from peer review of code through rOpenSci. Special thanks to Julia Romanowska ([jromanowska](https://github.com/jromanowska)) and Eric Brown ([eebrown](https://github.com/eebrown)) for reviewing our code and giving helpful suggestions on how to improve the clarity of workloopR's presentation.

> We are also happy to share that a journal article that accompanies this package was also peer reviewed and accepted by the [Journal of Open Source Software](https://joss.theoj.org/) (JOSS); our paper can be found [here](https://doi.org/10.21105/joss.01856).

Reviewer: ```suggestion
We are also happy to share that a journal article that accompanies this package was also peer reviewed and accepted by the Journal of Open Source Software (JOSS)[^2].
```

turned paper link to citation format

🐢

> [^1]: Software from Aurora Scientific: https://aurorascientific.com/products/muscle-physiology/muscle-software/

Reviewer: ```suggestion
[^1]: Software from Aurora Scientific: https://aurorascientific.com/products/muscle-physiology/muscle-software/
[^2]: Baliga V. and Senthivasan S (2019). workloopR: Analysis of work loops and other data from muscle physiology experiments in R. Journal of Open Source Software, 4(43), 1856, https://doi.org/10.21105/joss.01856
```

### How to get workloopR

We are not (yet) on CRAN but the package is available through rOpenSci's GitHub:
``` r
#install.packages("devtools") # if devtools is not installed
devtools::install_github("ropensci/workloopR")

## To build vignettes as well:
devtools::install_github("ropensci/workloopR", build_vignettes = TRUE)
```

### Feel free to make suggestions or requests

workloopR, as the name implies, was originally designed to handle data from work loop experiments as well as experiments that are complementary to work loops like twitch and tetanic trials. 

We are happy to expand the scope of the package to incorporate even more types of analyses of muscle physiology or biomechanics. We're presently eager to add data import functions for non-`ddf` file types. We're also interested in integrating support for electromyographic ([EMG](https://en.wikipedia.org/wiki/Electromyography)) data recorded directly from muscles, but that may take a little longer to develop.

Should you like to suggest a specific feature, please use the [Issues page](https://github.com/ropensci/workloopR/issues/new/choose) of our repository. 

### Package review by rOpenSci and JOSS

> We are thankful for the suggestions we've already received. workloopR benefited a lot from peer review of code through rOpenSci. Special thanks to Julia Romanowska ([jromanowska](https://github.com/jromanowska)) and Eric Brown ([eebrown](https://github.com/eebrown)) for reviewing our code and giving helpful suggestions on how to improve the clarity of workloopR's presentation.

Reviewer: ```suggestion
We are thankful for the suggestions we've already received. workloopR benefited a lot from [peer review of code](https://github.com/ropensci/software-review/issues/326) through rOpenSci. Special thanks to Julia Romanowska ([jromanowska](https://github.com/jromanowska)) and Eric Brown ([eebrown](https://github.com/eebrown)) for reviewing our code and giving helpful suggestions on how to improve the clarity of workloopR's presentation.
```
link to review thread
